src/api/archive_unstable_storage.md

```diff
@@ -3,91 +3,119 @@
-- `includeDescendants`: Boolean indicating whether the key-values of all the descendants of the `key` should be returned as well.
-
-**Return value**: String containing an opaque value representing the operation, or `null` if no block with that `hash` exists.
 
-The JSON-RPC server must obtain the value of the entry with the given `key` from the storage, either from the main trie of from `childTrie`. If `includeDescendants` is `true`, then the values of all the descendants must be obtained as well.
+Each element in `items` must be an object containing the following fields:
 
-If the block was previously returned by `archive_unstable_hashByHeight` at a height inferior or equal to the current finalized block height (as indicated by `archive_unstable_finalizedHeight`), then calling this method multiple times is guaranteed to always return non-null and always the same results.
-
-If the block was previously returned by `archive_unstable_hashByHeight` at a height strictly superior to the current finalized block height (as indicated by `archive_unstable_finalizedHeight`), then the block might "disappear" and calling this function might return `null` at any point.
+- `key`: String containing the hexadecimal-encoded key to fetch in the storage.
+- `type`: String equal to one of: `value`, `hash`, `closestDescendantMerkleValue`, `descendantsValues`, `descendantsHashes`.
+- `paginationStartKey`: This parameter is optional and should be a string containing the hexadecimal-encoded key from which the storage iteration should resume. This parameter is only valid in the context of `descendantsValues` and `descendantsHashes`.
 
-## Notifications format
+**Return value**: A JSON object.
 
-This function will later generate notifications in the following format:
+The JSON object returned by this function has the following format:
 
-```json
+```
 {
-    "jsonrpc": "2.0",
-    "method": "archive_unstable_storageEvent",
-    "params": {
-        "subscription": "...",
-        "result": ...
-    }
+    "result": [
+        {
+            ...
+        },
+        ...
+    ],
+    "discardedItems": ...
 }
 ```
 
-Where `subscription` is equal to the value returned by this function, and `result` can be one of:
+Where:
+
+- `result` contains a vector of JSON objects (possibly empty) that were found in the storage.
+- `discardedItems` is an integer indicating the number of items at the back of the array of the `items` parameters that couldn't be processed.
+
+The JSON objects in the `"result"` field can have one of the following formats based on their type:
 
-### item
+### Value
 
-```json
+```
 {
-    "event": "item",
     "key": "0x0000000...",
     "value": "0x0000000...",
 }
 ```
 
-Yields an item that was found in the storage.
+The JSON object corresponds to one of the requested items whose `type` was `"value"` or `"descendantsValues"`.
+
+If the `key` is not associated with a storage value in the trie, then no response is generated.
```

Comment:
I guess this is fine but would be easier for clients if the server would just send back `null` to indicate that key was not found.

For this API, a client probably needs to have a timeout for these request and after for instance 1 min the call is regarded as failed.

Reply:
Will modify the phrasing of the spec. What I meant by this, given archive/storage is a plain method, no response will be generated for this "key" in the "result" array. Users will still get back once all queries are processed the responses back